src/VarCache.ts

```diff
@@ -99,6 +98,10 @@ export default class VarCache {
     }
   }
 
+  public getVariablesChangedHandlers() {
+    return this.variablesChangedHandlers
+  }
+
```

Comment:
This method is used only for testing, and it exposes an implementation detail. This makes it hard to refactor the implementation, even if the behavior is not affected.

Instead of exposing the handlers, just make sure that added handlers are called and removed are not. This tests the event registration, without knowing how it works. This allows the implementation to be extracted in the future (as it is also needed for the start handler, and possibly about other handlers, e.g. `showMessage`)

Reply:
Thanks, updated as suggested!